logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/UserSearchApiWrapper.kt

```diff
@@ -36,36 +50,67 @@ internal class UserSearchApiWrapperImpl(
         domain: String,
         maxResultSize: Int?,
         searchUsersOptions: SearchUsersOptions
-    ): Either<NetworkFailure, UserSearchResponse> {
-
-        val searchResponse = wrapApiRequest {
+    ): Either<NetworkFailure, UserSearchResponse> =
+        wrapApiRequest {
             userSearchApi.search(
                 UserSearchRequest(
                     searchQuery = searchQuery,
                     domain = domain,
                     maxResultSize = maxResultSize
                 )
             )
+        }.map { userSearchResponse ->
+            filter(userSearchResponse, searchUsersOptions)
         }
 
-        return if (searchUsersOptions.conversationExcluded is ConversationMemberExcludedOptions.ConversationExcluded) {
-            val conversationMembersId = conversationDAO.getAllMembers(
-                qualifiedID = idMapper.toDaoModel(qualifiedID = searchUsersOptions.conversationExcluded.conversationId)
-            ).firstOrNull()?.map { idMapper.fromDaoModel(it.user) }
+    private suspend fun filter(
+        userSearchResponse: UserSearchResponse,
+        searchUsersOptions: SearchUsersOptions
+    ): UserSearchResponse {
+        val selfUser = getSelfUser()
+
+        val filteredContactResponse =
+            if (searchUsersOptions.conversationExcluded is ConversationMemberExcludedOptions.ConversationExcluded) {
+                val conversationMembersId = conversationDAO.getAllMembers(
+                    qualifiedID = idMapper.toDaoModel(qualifiedID = searchUsersOptions.conversationExcluded.conversationId)
+                ).firstOrNull()?.map { idMapper.fromDaoModel(it.user) }
 
-            searchResponse.map {
-                val filteredContactResponse = it.documents.filter { contactDTO ->
-                    !(conversationMembersId?.contains(idMapper.fromApiModel(contactDTO.qualifiedID)) ?: false)
+                userSearchResponse.documents.filter { contactDTO ->
+                    val domainId = idMapper.fromApiModel(contactDTO.qualifiedID)
+
+                    val isConversationMember = conversationMembersId?.contains(domainId) ?: false
+                    val isSelfUser = selfUser.id == domainId
+
+                    // negate it because that is exactly what we do not want to have in filter results
+                    !(isConversationMember || isSelfUser)
```

Comment:
Maybe extract those to separate functionality and name it instead of commenting for better readability?

Reply:
in my opinion without comment it should be clear enough :P I added the comment just to make it even more clear enough I think a private function would be a overkill here